src/main/java/org/jabref/gui/externalfiles/UnlinkedFilesDialogView.java

```diff
@@ -0,0 +1,264 @@
+import org.controlsfx.control.CheckTreeView;
+
+public class UnlinkedFilesDialogView extends BaseDialog<Void> {
+
+    @FXML private TextField directoryPathField;
+    @FXML private ComboBox<FileExtensionViewModel> fileTypeCombo;
+    @FXML private CheckTreeView<FileNodeViewModel> unlinkedFilesList;
+    @FXML private Button scanButton;
+    @FXML private Button exportButton;
+    @FXML private Button importButton;
+    @FXML private Label progressText;
+    @FXML private Accordion accordion;
+    @FXML private ProgressIndicator progressDisplay;
+    @FXML private VBox progressPane;
+
+    @FXML private TableView<ImportFilesResultItemViewModel> importResultTable;
+    @FXML private TableColumn<ImportFilesResultItemViewModel, JabRefIcon> colStatus;
+    @FXML private TableColumn<ImportFilesResultItemViewModel, String> colMessage;
+    @FXML private TableColumn<ImportFilesResultItemViewModel, String> colFile;
+
+    @FXML private TitledPane filePane;
+    @FXML private TitledPane resultPane;
+
+    @Inject private PreferencesService preferencesService;
+    @Inject private DialogService dialogService;
+    @Inject private StateManager stateManager;
+    @Inject private UndoManager undoManager;
+    @Inject private TaskExecutor taskExecutor;
+    @Inject private FileUpdateMonitor fileUpdateMonitor;
+
+    private final ControlsFxVisualizer validationVisualizer;
+    private UnlinkedFilesDialogViewModel viewModel;
+
+    public UnlinkedFilesDialogView() {
+        this.validationVisualizer = new ControlsFxVisualizer();
+
+        this.setTitle(Localization.lang("Search for unlinked local files"));
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        setResultConverter(button -> {
+            if (button == ButtonType.CANCEL) {
+                viewModel.cancelTasks();
+            }
+            return null;
+        });
+    }
+
+    @FXML
+    private void initialize() {
+        viewModel = new UnlinkedFilesDialogViewModel(dialogService, ExternalFileTypes.getInstance(), undoManager, fileUpdateMonitor, preferencesService, stateManager, taskExecutor);
+
+        progressDisplay.progressProperty().bind(viewModel.progressValueProperty());
+        progressText.textProperty().bind(viewModel.progressTextProperty());
+        progressPane.managedProperty().bind(viewModel.taskActiveProperty());
+        progressPane.visibleProperty().bind(viewModel.taskActiveProperty());
+        accordion.disableProperty().bind(viewModel.taskActiveProperty());
+
+        viewModel.treeRootProperty().addListener(observable -> {
+            scanButton.setDefaultButton(false);
+            importButton.setDefaultButton(true);
+            scanButton.setDefaultButton(false);
+            filePane.setExpanded(true);
+            resultPane.setExpanded(false);
+        });
+
+        viewModel.resultTableItems().addListener((InvalidationListener) observable -> {
```

Comment:
A bit cleaner: bind to `hasNoResults = EasyBind.isEmpty(viewModel.resultTableItems())` (similar for the treeRoot listener above)

Reply:
I tried with Easybind.subscribe and Bindings.isEmpty, but that did not trigger correctly

Reply:
Idea was always to open the pane if a change happens.